cmd/kubeapps-apis/plugins/fluxv2/packages/v1alpha1/release.go

```diff
@@ -491,17 +500,83 @@ func (s *Server) deleteRelease(ctx context.Context, packageRef *corev1.Installed
 	log.V(4).Infof("Deleted release: [%s]", packageRef.Identifier)
 
 	if err = ifc.Delete(ctx, packageRef.Identifier, metav1.DeleteOptions{}); err != nil {
-		if errors.IsNotFound(err) {
-			return status.Errorf(codes.NotFound, "%q", err)
-		} else if errors.IsForbidden(err) || errors.IsUnauthorized(err) {
-			return status.Errorf(codes.Unauthenticated, "Unable to delete release due to %v", err)
-		} else {
-			return status.Errorf(codes.Internal, "Unable to delete release due to %v", err)
-		}
+		return statuserror.FromK8sError("delete", "HelmRelease", packageRef.Identifier, err)
 	}
 	return nil
 }
 
+// Potentially, there are 3 different namespaces that can be specified here
+// 1. spec.chart.spec.sourceRef.namespace, where HelmRepository CRD object referenced exists
+// 2. metadata.namespace, where this HelmRelease CRD will exist, same as (3) below
+//    per https://github.com/kubeapps/kubeapps/pull/3640#issuecomment-949315105
+// 3. spec.targetNamespace, where flux will install any artifacts from the release
+func (s *Server) newFluxHelmRelease(chart *models.Chart, targetName types.NamespacedName, versionRef *corev1.VersionReference, reconcile *corev1.ReconciliationOptions, values map[string]interface{}) (*unstructured.Unstructured, error) {
+	unstructuredRel := unstructured.Unstructured{
```

Comment:
yes, this has been on my TODO list ever since you mentioned it first a couple months ago). Just need to confirm with Pepe I have bandwidth to do this.